### PR TITLE
[flang] Do not create .f18.mod files for each compiled module

### PR DESCRIPTION
**Pull request by mjklemm**
The default CMake scripts had a copy operation to copy a compiled `.mod` file to also be available with suffix `.f18.mod`.  This seems no longer needed.  Also updated ModFiles.md to point to `-module-suffix`.

### CODE REVIEW

File: flang/docs/ModFiles.md [hunk 1]
```diff
@@ -27,7 +27,9 @@ often use `rm *.mod` to clean up.
 The disadvantage of using the same name as other compilers is that it is not
 clear which compiler created a `.mod` file and files from multiple compilers
 cannot be in the same directory. This could be solved by adding something
-between the module name and extension, e.g. `<modulename>-f18.mod`.
+between the module name and extension, e.g. `<modulename>-f18.mod`.  If this
+needed, Flang's fc1 accepts the option `-module-suffix` to alter the suffix
+used for the module file on disk.
```
Comment (kkwli) [inline flang/docs/ModFiles.md:32]: I don't think 'on disk' is necessary here.

**Review by kkwli — APPROVED** [review]
LG

Thanks.

File: flang/tools/f18/CMakeLists.txt [hunk 2]
```diff
@@ -47,7 +47,7 @@ if (NOT CMAKE_CROSSCOMPILING)
       endif()
     endif()
 
-    # The module contains PPC vector types that needs the PPC target.
+    # The module contains PPC vector types that needs th1e PPC target.
```
[the remark this answers is not in the document]
Comment (mjklemm) [reply flang/tools/f18/CMakeLists.txt:50]: Fixed.

File: flang/docs/ModFiles.md (hunk 1 above)
Comment (mjklemm) [reply flang/docs/ModFiles.md:32]: Agreed.  Removed.

**Comment by mjklemm** [comment]
@kiranchandramohan @banach-space Ping :-)

**Review by kiranchandramohan — COMMENTED** [review]
Would the flang-to-external-fc script (https://flang.llvm.org/docs/FlangDriver.html#the-flang-to-external-fc-script) work without this? The script might not be needed if nobody is using it. If so it might be good to remove that in a separate patch.

**Comment by mjklemm** [comment]
> Would the flang-to-external-fc script (https://flang.llvm.org/docs/FlangDriver.html#the-flang-to-external-fc-script) work without this? The script might not be needed if nobody is using it. If so it might be good to remove that in a separate patch.

I have touch-tested this and the `flang-to-external-fc` seems to have worked and compiled some simple tests.

**Review by banach-space — APPROVED** [review]
LG, thanks!

> Would the flang-to-external-fc script (https://flang.llvm.org/docs/FlangDriver.html#the-flang-to-external-fc-script) work without this? The script might not be needed if nobody is using it. If so it might be good to remove that in a separate patch.

+1 to removing it, but it would be good to ask on Discourse first

**Pull request merged**